src/bin/pgcopydb/summary.c

```diff
@@ -2347,21 +2322,29 @@ catalog_timing_fetch(SQLiteQuery *query)
 	timing->doneTime = sqlite3_column_int64(query->ppStmt, 3);
 	timing->durationMs = sqlite3_column_int64(query->ppStmt, 4);
 
-	if (sqlite3_column_type(query->ppStmt, 5) != SQLITE_NULL)
+	/*
+	 * Skip reading `ppDuration` from DB and create it from `durationMs`
+	 * in accordance with the single source of truth principle.
+	 */
```

Comment:
This principle was already implemented before, where the decision was made to compute the data before SQLite storage and then use what is in the SQLite database. This allows any external tooling to have access to the same data as our internal code. Including commands that run alongside the main `pgcopydb` process rather than as a child process.

IOW I believe you introduced a regression here.

Reply:
The problem was that the data in these 2 fields were inconsistent. What are the external tools which use these fields?

Reply:
- Is there a way to fix the bug wrt inconsistencies here, and keeping the data all in the SQLite database file? 
- Could the inconsistency be an explanation for https://github.com/dimitri/pgcopydb/issues/697?
- I mean “external” as in “not running within the process-tree with access to all the memory etc”, so that means the command `pgcopydb list progress` is already an “external” tool. All it has access to is the SQLite database file contents.

Reply:
- Probably there is. But is it a good idea to keep the 2 versions of the same data in the DB?
- I don't know, maybe.
- If we are talking about the `pgcopydb` then that means we have no problem here.

Reply:
Now that we use our own version of SQLite and we know we have JSON support in SQL, we could rewrite all the SQLite to JSON transformation code and use a SQL query instead. It was my impression that having the data in two forms in the SQLite database was useful for that.

Another use-case is debugging from the SQLite data file when something strange happens. Then we might appreciate having the data available in pretty printed format for debugging.

The use-case I mentioned before still do exists, it's nice to be able to re-use the data from the SQLite data file either in numeric format or in pretty-printed format without having to re-compute the pretty-printed format every time. It's always the same with caching: do you want to use more CPU or more memory?